pkg/agent/route/route_linux.go

```diff
@@ -2589,3 +2638,185 @@ func (c *Client) DeleteNodeNetworkPolicyIPTables(iptablesChains []string, isIPv6
+					"ip", "daddr", "@", antreaNFTablesSetPeerPodCIDR,
+					"flow", "add", "@", antreaNFTablesFlowtable,
+					"counter",
+				),
+				Comment: ptr.To("Accelerate IPv4 connections: local Pod CIDR to remote Pod CIDRs"),
+			})
+			tx.Add(&knftables.Rule{
+				Chain: antreaNFTablesChainForward,
+				Rule: knftables.Concat(
+					"iif", c.nodeConfig.NodeTransportInterfaceName,
+					"ip", "saddr", "@", antreaNFTablesSetPeerPodCIDR,
+					"oif", c.nodeConfig.GatewayConfig.Name,
+					"ip", "daddr", c.nodeConfig.PodIPv4CIDR.String(),
+					"flow", "add", "@", antreaNFTablesFlowtable,
+					"counter",
+				),
+				Comment: ptr.To("Accelerate IPv4 connections: remote Pod CIDRs to local Pod CIDR"),
+			})
+		}
+		if ipProtocol == knftables.IPv6Family {
+			// Add the nft set for storing peer Pod IPv6 CIDRs.
+			tx.Add(&knftables.Set{
+				Name:    antreaNFTablesSetPeerPodCIDR,
+				Type:    "ipv6_addr",
+				Flags:   []knftables.SetFlag{knftables.IntervalFlag},
+				Comment: ptr.To("Set for storing IPv6 peer Pods CIDRs"),
+			})
+			// Flush the set if it exists.
+			tx.Flush(&knftables.Set{
+				Name: antreaNFTablesSetPeerPodCIDR,
+			})
+			// Restore existing IPv6 peer Pod CIDRs into the set.
+			c.podCIDRNFTablesSetIPv6.Range(func(_, v interface{}) bool {
+				return func(element *knftables.Element) bool {
+					tx.Add(element)
+					return true
+				}(v.(*knftables.Element))
+			})
+			// Add rules to accelerate Pod-to-Pod IPv6 connections via flowtable.
+			tx.Add(&knftables.Rule{
+				Chain: antreaNFTablesChainForward,
+				Rule: knftables.Concat(
+					"iif", c.nodeConfig.GatewayConfig.Name,
+					"ip6", "saddr", c.nodeConfig.PodIPv6CIDR.String(),
+					"oif", c.nodeConfig.NodeTransportInterfaceName,
+					"ip6", "daddr", "@", antreaNFTablesSetPeerPodCIDR,
+					"flow", "add", "@", antreaNFTablesFlowtable,
+					"counter",
+				),
+				Comment: ptr.To("Accelerate IPv6 connections: local Pod CIDR to remote Pod CIDRs"),
+			})
+			tx.Add(&knftables.Rule{
+				Chain: antreaNFTablesChainForward,
+				Rule: knftables.Concat(
+					"iif", c.nodeConfig.NodeTransportInterfaceName,
+					"ip6", "saddr", "@", antreaNFTablesSetPeerPodCIDR,
+					"oif", c.nodeConfig.GatewayConfig.Name,
+					"ip6", "daddr", c.nodeConfig.PodIPv6CIDR.String(),
+					"flow", "add", "@", antreaNFTablesFlowtable,
+					"counter",
+				),
+				Comment: ptr.To("Accelerate IPv6 connections: remote Pod CIDRs to local Pod CIDR"),
+			})
+		}
+
+		if err := nft.Run(context.TODO(), tx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) addPeerPodCIDRToNFTablesSet(podCIDR *net.IPNet) error {
+	var nft knftables.Interface
+	isIPv6 := utilnet.IsIPv6(podCIDR.IP)
+	if isIPv6 {
+		nft = c.nftables.IPv6
+	} else {
+		nft = c.nftables.IPv4
+	}
+
+	tx := nft.NewTransaction()
+	element := &knftables.Element{
+		Set: antreaNFTablesSetPeerPodCIDR,
+		Key: []string{podCIDR.String()},
+	}
+	tx.Add(element)
+	if err := nft.Run(context.TODO(), tx); err != nil {
```

Comment:
a general comment for the `nft.Run` calls: do you think they would benefit from having a reasonable timeout or from having the caller provide an actual context? Is there any reason that could cause the command to hang indefinitely or for a long time?

Reply:
No file lock is used in nftables like the file lock used by iptables. It's safe to call `nft -f <filename>` concurrently to update a table because there is a kernel-side lock `nfnl_lock` to protect the update processing.